src/components/Collection/Series/SeriesEpisode.tsx

```diff
@@ -57,61 +57,63 @@ const SeriesEpisode = ({ animeId, episode, nextUp, page }: Props) => {
   const handleMarkWatched = useEventCallback(() => markWatched({ episodeId, watched: episode.Watched === null }));
   const handleMarkHidden = useEventCallback(() => markHidden({ episodeId, hidden: !episode.IsHidden }));
 
+  const filesExist = useMemo(() => episode.Size > 0, [episode.Size]);
+
   return (
     <>
       <div className={cx('z-10 flex items-center gap-x-8', !nextUp && 'p-8')}>
         <BackgroundImagePlaceholderDiv
           image={thumbnail}
           className="group h-[13rem] min-w-[22.3125rem] rounded-md border border-panel-border"
-          hidePlaceholderOnHover
+          hidePlaceholderOnHover={filesExist}
           zoomOnHover
         >
-          <div className="pointer-events-none absolute right-3 top-3 z-10 transition-opacity group-hover:opacity-0">
-            <StateIcon icon={mdiEyeCheckOutline} show={episode.Watched !== null} />
-            <StateIcon icon={mdiEyeOffOutline} show={episode.IsHidden} />
-          </div>
-          <div className="pointer-events-none z-10 flex h-full justify-between bg-panel-background-transparent p-3 opacity-0 transition-opacity group-hover:pointer-events-auto group-hover:opacity-100">
-            <div>
-              <StateButton icon={mdiPencilCircleOutline} active={false} onClick={() => {}} />
-            </div>
-            <div className="flex flex-col gap-y-8">
-              <StateButton
-                icon={mdiEyeCheckOutline}
-                active={episode.Watched !== null}
-                onClick={handleMarkWatched}
-              />
-              <StateButton
-                icon={mdiEyeOffOutline}
-                active={episode.IsHidden}
-                onClick={handleMarkHidden}
-              />
-            </div>
-          </div>
+          {filesExist && (
+            <>
+              <div className="pointer-events-none absolute right-3 top-3 z-10 transition-opacity group-hover:opacity-0">
+                <StateIcon icon={mdiEyeCheckOutline} show={episode.Watched !== null} />
+                <StateIcon icon={mdiEyeOffOutline} show={episode.IsHidden} />
+              </div>
+              <div className="pointer-events-none z-10 flex h-full justify-between bg-panel-background-transparent p-3 opacity-0 transition-opacity group-hover:pointer-events-auto group-hover:opacity-100">
+                <div>
+                  <StateButton icon={mdiPencilCircleOutline} active={false} onClick={() => {}} />
+                </div>
+                <div className="flex flex-col gap-y-8">
+                  <StateButton
+                    icon={mdiEyeCheckOutline}
+                    active={episode.Watched !== null}
+                    onClick={handleMarkWatched}
+                  />
+                  <StateButton
+                    icon={mdiEyeOffOutline}
```

Comment:
@harshithmohan you know you just broke hiding "missing" episodes you don'twant to show up... right?